### PR TITLE
Mtdsa 23605 - CreateAmendCgtPpdOverrides

### PR DESCRIPTION
**Pull request by BhaveshParmar00**

### CODE REVIEW

File: test/v1/controllers/validators/CreateAmendCgtPpdOverridesValidatorFactorySpec.scala [hunk 1]
```diff
@@ -551,193 +555,301 @@ class CreateAmendCgtPpdOverridesValidatorSpec extends UnitSpec with ValueFormatE
...
+        val result: Either[ErrorWrapper, CreateAmendCgtPpdOverridesRequestData] =
+          validator(validNino, validTaxYear, validRequestJson).validateAndWrapResult()
+        result shouldBe Right(CreateAmendCgtPpdOverridesRequestData(parsedNino, parsedTaxYear,parsedValidRequestBody))
       }
+    }
 
-      "a valid request contains only single disposals is supplied" in new Test {
-        validator.validate(
-          CreateAmendCgtPpdOverridesRawData(validNino, validTaxYear, AnyContentAsJson(validOnlySinglePropertyDisposalsRequestJson))) shouldBe Nil
+      "a valid request contains only multiple disposals is supplied" in  {
```
Comment (KobithasanVasantharajah) [inline test/v1/controllers/validators/CreateAmendCgtPpdOverridesValidatorFactorySpec.scala:585]: Shouldn't this read: 
```suggestion
      "a valid request containing only multiple disposals is supplied" in  {
```

File: test/v1/controllers/validators/CreateAmendCgtPpdOverridesValidatorFactorySpec.scala [hunk 2]
```diff
@@ -551,193 +555,301 @@ class CreateAmendCgtPpdOverridesValidatorSpec extends UnitSpec with ValueFormatE
...
+          validator(validNino, validTaxYear, validOnlyMultiplePropertyDisposalsRequestJson).validateAndWrapResult()
+        result shouldBe Right(CreateAmendCgtPpdOverridesRequestData(parsedNino, parsedTaxYear,parsedValidMultipleOnlyBody))
       }
-    }
 
-    "return NinoFormatError error" when {
-      "an invalid nino is supplied" in new Test {
-        validator.validate(CreateAmendCgtPpdOverridesRawData("A12344A", validTaxYear, validRequestBody)) shouldBe
-          List(NinoFormatError)
+      "a valid request contains only single disposals is supplied" in  {
```
Comment (KobithasanVasantharajah) [inline test/v1/controllers/validators/CreateAmendCgtPpdOverridesValidatorFactorySpec.scala:591]: Shouldn't this read:
```suggestion
      "a valid request containing only single disposals is supplied" in  {
```

File: test/v1/controllers/validators/CreateAmendCgtPpdOverridesValidatorFactorySpec.scala [hunk 3]
```diff
@@ -551,193 +555,301 @@ class CreateAmendCgtPpdOverridesValidatorSpec extends UnitSpec with ValueFormatE
...
+              "/singlePropertyDisposals/1/amountOfNetLoss"
             ))
+          )
+        )
       }
     }
 
     "return a dateFormatError" when {
-      "a body with an incorrect date is provided" in new Test { // 20-02-28 //2020-03-29
+      "a body with an incorrect date is provided" in { // 20-02-28 //2020-03-29
```
Comment (KobithasanVasantharajah) [inline test/v1/controllers/validators/CreateAmendCgtPpdOverridesValidatorFactorySpec.scala:750]: Can we remove the inline comment here, please?

**Review by KobithasanVasantharajah — CHANGES_REQUESTED** [review]

File: app/api/models/errors/CommonMtdErrors.scala [hunk 4]
```diff
@@ -52,6 +52,8 @@ object ClaimOrElectionCodesFormatError
 
 object TransactionIdFormatError extends MtdError(code = "FORMAT_TRANSACTION_ID", message = "The transaction ID format is invalid", BAD_REQUEST)
 
+object StringFormatError extends MtdError(code = "FORMAT_STRING", message = "The supplied string format is not valid", BAD_REQUEST)
```
[the remark this answers is not in the document]
Comment (BhaveshParmar00) [reply app/api/models/errors/CommonMtdErrors.scala:55]: Added whilst testing, unused error - removed now

**Review by KobithasanVasantharajah — APPROVED** [review]

**Review by KobithasanVasantharajah — APPROVED** [review]

**Pull request merged**